src/VisualStudio/Core/Def/Watson/FaultReporter.cs

```diff
@@ -195,57 +201,81 @@ private static string GetDescription(Exception exception)
             return exception.Message;
         }
 
-        private static List<string> CollectServiceHubLogFilePaths()
+        private static IList<string> CollectLogHubFilePaths()
         {
-            var paths = new List<string>();
+            try
+            {
+                var logPath = Path.Combine(Path.GetTempPath(), "VSLogs");
+                var logs = CollectFilePaths(logPath, "*.svclog", (name) => !name.Contains("Roslyn") && !name.Contains("LSPClient"));
+                return logs;
+            }
+            catch (Exception)
+            {
+                // ignore failures
+            }
 
+            return SpecializedCollections.EmptyList<string>();
+        }
+
+        private static IList<string> CollectServiceHubLogFilePaths()
+        {
             try
             {
                 var logPath = Path.Combine(Path.GetTempPath(), "servicehub", "logs");
-                if (!Directory.Exists(logPath))
-                {
-                    return paths;
-                }
 
-                // attach all log files that are modified less than 1 day before.
-                var now = DateTime.UtcNow;
-                var oneDay = TimeSpan.FromDays(1);
+                // TODO: https://github.com/dotnet/roslyn/issues/42582 
+                // name our services more consistently to simplify filtering
+                var logs = CollectFilePaths(logPath, "*.log", (name) => !name.Contains("-" + ServiceDescriptor.ServiceNameTopLevelPrefix) &&
```

Comment:
Small nit: consider putting the named parameter for the lambda, so it's clear this is an exclude filter vs. an include filter. (Although a lambda where everything is a ! makes me wonder if an include filter is saner here.)